### PR TITLE
Enforce strict & warnings for tests

### PR DESCRIPTION
**Pull request by atoomic**
Make sure all tests are running with strict and warnings enabled.
Also use Sub::Util to define prototypes.

Note: this is using #117 to make sure the test suite is clean

view: https://github.com/atoomic/pod-simple/actions/runs/141064668

### CODE REVIEW

**Comment by atoomic** [comment]
We should make sure this is passing using 5.6.2 and indirect calls is the way to go...

**Comment by jkeenan** [comment]
The first production release of the Perl 5 core distribution which shipped with Pod::Simple was 5.10.0.  The first production release which shipped with Sub::Util was 5.22.0 (per my reading of dist/Module-CoreList/lib/Module/CoreList.pm).  What does this merge commit imply for perl users on versions 5.10 through 5.20?

(I ask because prototype-signature confusion is the single most daunting problem we're facing in the more inclusive variant of Perl 7, found in the `p7` branch [here](https://github.com/atoomic/perl).

Thank you very much.
Jim Keenan

File: t/JustPod_corpus.t [hunk 1]
```diff
@@ -58,99 +58,129 @@ BEGIN {
...
-
-  if ($parser->any_errata_seen()) {
-    pass("Skip '$file' because of pod errors");
-    next if "$]" lt '5.010.001';     # note() not found in earlier versions
-    my $errata = $parser->errata_seen();
-    foreach my $line_number (sort { $a <=> $b } keys %$errata) {
-        foreach my $err_msg (sort @{$errata->{$line_number}}) {
-            note("$file: $line_number: $err_msg");
-        }
+  SKIP: {
```
Comment (karenetheridge) [inline t/JustPod_corpus.t:87]: Doing `skip(..) and next if $conditions` would mean the SKIP block isn't needed, which removes the extra indentation and the whitespace changes on every line.

**Comment by karenetheridge** [comment]
>  What does this merge commit imply for perl users on versions 5.10 through 5.20?

Sub::Util will need to be added as an explicit prereq in Makefile.PL. (It ought to be listed anyway, regardless, but since it isn't in core for all versions that Pod::Simple is, it *needs* to be.)

File: t/JustPod_corpus.t (hunk 1 above)
Comment (atoomic) [reply t/JustPod_corpus.t:87]: not sure what's your complain here, there is `Hide whitespace change` at the top
these tests were broken under windows and need investigation, but this was outside the scope of this change

File: Makefile.PL [hunk 2]
```diff
@@ -9,63 +9,66 @@
...
-  VERSION_FROM	=> 'lib/Pod/Simple.pm',
-  ABSTRACT_FROM	=> 'lib/Pod/Simple.pod',
-  PREREQ_PM	  	=> {
-   'Text::Wrap' => '98.112902',
-   'Pod::Escapes' => '1.04',
+    NAME          => 'Pod::Simple',
+    VERSION_FROM  => 'lib/Pod/Simple.pm',
+    ABSTRACT_FROM => 'lib/Pod/Simple.pod',
+    TEST_REQUIRES => {
+        'Sub::Util' => 1.55,
```
Comment (atoomic) [inline Makefile.PL:19]: @karenetheridge notice the extra requirement for `Sub-Util`
are you saying we need it also as a regular `PREREQ_PM` even if it's only use by unit tests?

Comment (karenetheridge) [reply Makefile.PL:19]: ok great; I missed the test_requires update.

File: t/JustPod_corpus.t (hunk 1 above)
Comment (karenetheridge) [reply t/JustPod_corpus.t:87]: This is not a complaint.

I'm observing that a number of lines are changing only in whitespace, which are nice to avoid when possible. This affects functionality like 'git blame'.

Comment (atoomic) [reply t/JustPod_corpus.t:87]: ok I see thanks for this precision.

**Comment by khwilliamson** [comment]
The policy is to do white space changes in a separate commit, if there are many of them.  I use git blame -w to find real changes.

Is this ready for release? I don't see @karenetheridge point about 

> Sub::Util will need to be added as an explicit prereq in Makefile.PL
being addressed

**Comment by atoomic** [comment]
@khwilliamson also note that GitHub UI provides a `Hide whitespace changes` option when clicking on the wheel on the top of a PR

File: Makefile.PL [hunk 3]
```diff
@@ -9,63 +9,66 @@
 require 5;
 
 use strict;
-use ExtUtils::MakeMaker;
+use ExtUtils::MakeMaker 6.64;
```
Comment (karenetheridge) [inline Makefile.PL:12]: could the Makefile.PL updates be moved to a separate commit? Other than adding Sub::Util as a prereq, these changes don't look to be related to the change in prototypes.

File: t/begin.t [hunk 4]
```diff
@@ -18,7 +19,13 @@ ok 1;
 use Pod::Simple::DumpAsXML;
 use Pod::Simple::XMLOutStream;
 print "# Pod::Simple version $Pod::Simple::VERSION\n";
-sub e ($$) { Pod::Simple::DumpAsXML->_duo(@_) }
+
+BEGIN {
+  require FindBin;
```
Comment (karenetheridge) [inline t/begin.t:24]: Is this necessary? Surely a simple `use lib 't/lib'; use helpers;` should be fine here.

**Comment by karenetheridge** [comment]
Also, this PR should be rebased against master before being merged. (It can be done by the author and force-pushed, but can also be done on the command line by the merger.)

**Comment by karenetheridge** [comment]
Is Sub::Util 1.55 needed? I see `set_prototype` was added in 1.40.

**Comment by jkeenan** [comment]
This p.r. was closed by @atoomic on July 16.  Should it be re-opened?  If not, why are we still discussing it.

Signed,
Confused

**Comment by khwilliamson** [comment]
I hadn't realized this PR was superseded by #119 .  So that addressed the Sub:Util issue, but did it?  We should move conversation to that.
